en/05_Contributing/05_Coding_Conventions/01_PHP_Coding_Conventions.md

```diff
@@ -136,6 +136,133 @@ class MyClass extends Class
+- Avoid writing any new "magic" code as it is very hard to follow.
+- Within reason, favour larger files that have everything contained in them rather than multiple small files that require developers to jump around multiple files to follow a single code path.
+
+## Inline comments
+- If there is a non-obvious reason for a code operation, include verbose inline comments that explain *why* we are doing this operation. Remember that while something may be obvious to you now, it won't be obvious to someone else in 6 months.
+- Inline comments that explain *what* the code is doing are generally fine as it can make code easier to follow and is particularly useful for larger methods, though it may be better to simply split larger methods in to several private methods.
+
+## Naming conventions
+
+### Casing
+
+- Use `camelCase` for variables, method names and parameters
+- Use `UpperCamelCase` for class names, interfaces, traits and enums
+- Use `snake_case` for private static configuration properties.
+- Use `SCREAMING_SNAKE_CASE` for constants including class contants, and environment variables
+- Note that there will be some inconsistencies with legacy code where static methods have been a combination of `MyClass::camelCase()` and `MyClass::snake_case()`
+
+### Accessors
+
+- For accessors omit the leading 'get' with getters and prefix with 'set' for setters. e.g. `function myValue():sting { return $this->value; }` and `function setMyValue(string $value): void { $this->value = $value; }`.
+- Note that there will be some inconsistencies with legacy code as get accessors have been a combination of "with get", "without get", and "without get with the first letter uppercased"
+
+### File suffixes and prefixes
+
+- This is for new classes and interfaces that are created. Legacy Silvesrtripe code is inconsistent whether or not it follows these conventions.
+- Suffix the classnames with the type of class that is subclassed. The suffixed is sometimes a shortened version of the parent class because it reads better while retaining easy comprehension. Here are some common examples with the parents classes in brackets
+  - `Exception` (Exception)
+  - `Controller` (Controller)
+  - `Extension` (Extension, DataExtension)
+  - `Admin` (ModelAdmin)
+  - `Handler` (RequestHandler)
+  - `Filter` (SearchFilter)
+  - `Form` (Form)
+  - `Field` (FormField)
+  - `Job` (QueuedJob)
+  - `Task` (BuildTask)
+  - `Service` (generic class type that has no parent class)
+Some parent classes have unorthodox naming, in which case suffix with something more standard
+  - `Page` (SiteTree)
+  - `Controller` (LeftAndMain, CMSMain)
+- Do not suffix classes if they are a subclass any of the following:
+  - `DataObject`
+- Suffix interfaces with `Interface`
+- Suffix traits with `Trait`
+- Prefix abstract classes with `Abstract`
```

Comment:
It just makes it more obvious that it's a trait.

The other standard is to suffix it with 'able' e.g. Injectable, Composable, etc.  That also makes it obvious that it's a trait.  However we've only done that a few times

There's plrenty of stuff like `trait UploadReceiver` which simply doesn't sound trait like, though if it were called "UploadReceiverTrait" it's obvious that it's a trait

Reply:
> must be a verb followed by a noun, describing what it does 

Yeah, well you've got a very good grasp on the english language, others (I put myself in this camp), not so much. We'll forget rules like this.  Also many contributers are english as a second language.  We're likely to go around in circles arguing about the correct way to use the english language

>> ThrowsException

`trait ThrowException` - if I saw something suffixed with `Exception` my brain at first scan would mis-categorise this as an Exception, not a trait.

I've also recommended we suffix interfaces with `Interface`, suffixing with `Trait` is going to be consistent with that.  We also got a recommendation to suffix everything with 'Extension' so again it's consistent.

Reply:
The `ThrowsException` name was a bad example, I'll accept that. But I think the general concept is still good.

> We'll forget rules like this

That's fine, that's half the reason why we're documenting these standards. I think if there are some clear examples, then it should be pretty easy to see when someone is breaking the pattern.

> I've also recommended we suffix interfaces with `Interface`

I'm not super sure I like that either - but even so, suffixing `Trait` is only consistent with it in-so-far as suffixing `Class` is. But we don't suffix classes with `Class` because we have other rules about how to name them. If we have other rules about how to name traits, we don't need to suffix them with `Trait`.

> I just want things to require as little brain power as possible to comprehend

That's fine and a good goal. But we will still need to think of names for traits beyond the suffix. I don't really want traits named something like `ModelTrait` or `ControllerTrait` because we haven't given any guidance beyond sticking `Trait` on the end.


All of this said, if at the end of the day it's only you and I who care about this and nobody else chimes in, I'll accept suffixing traits with `Trait` since at least that's better than having _no guidance at all_ which is our current status.

Reply:
FWIW PSR has naming conventions on it's own contributions that includes prefixes/suffixes: https://www.php-fig.org/bylaws/psr-naming-conventions

My opinion is that the `Interface` and `Abstract` suffix/prefix are good. I personally prefer the `Trait` suffix too, but am not too fussed what's decided there.

Reply:
Removed bit about interfaces / abstract / traits